### PR TITLE
FEC-10554 Logic change for external/internal subtitles' selection

### PR DESCRIPTION
**Pull request by GouravSna**
Added new `PKSubtitlePreference`. 

`PKSubtitlePreference.INTERNAL, Internal will be present and External subtitle will be discarded
  PKSubtitlePreference.EXTERNAL, External will be present and Internal subtitle will be discarded
  PKSubtitlePreference.OFF, Both internal and external subtitles will be there`

Application needs to set the preference,
Example:  `player.getSettings().setSubtitlePreference(AppPreference);`

### CODE REVIEW

File: playkit/src/main/java/com/kaltura/playkit/PKSubtitlePreference.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.kaltura.playkit;
+
+public enum PKSubtitlePreference {
+    INTERNAL(1),
```
Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/PKSubtitlePreference.java]: why numbers are needed?

Comment (GouravSna) [reply playkit/src/main/java/com/kaltura/playkit/PKSubtitlePreference.java]: Removed it. Yes it was redundant.

File: playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java [hunk 2]
```diff
@@ -323,21 +324,30 @@ private String getExternalSubtitleLanguage(Format format) {
...
+            return false;
+        }
+
         String languageName = format.language;
         boolean isExternalSubtitle = isExternalSubtitle(format);
-        boolean isPreferInternalSubtitles = playerSettings.isPreferInternalSubtitles();
 
         if (isExternalSubtitle) {
             languageName = getExternalSubtitleLanguage(format);
+            if (subtitlePreference == PKSubtitlePreference.INTERNAL && !subtitleListMap.get(languageName).containsKey(languageName)) {
```
Comment (GouravSna) [inline playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java]: Check null first `subtitleListMap.get(languageName)` before checking for contains.

Comment (GouravSna) [reply playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java]: Done

**Review by giladna — APPROVED** [review]

**Pull request merged**